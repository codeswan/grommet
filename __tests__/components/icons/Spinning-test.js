// (C) Copyright 2014-2016 Hewlett Packard Enterprise Development LP

import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';

import Spinning from '../../../src/js/components/icons/Spinning';

describe('Spinning', () => {
  it('has correct default options', () => {
    const component = renderer.create(
      <Spinning />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct small=true rendering', () => {
    const component = renderer.create(
      <Spinning small={true} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('has correct className rendering', () => {
    const component = renderer.create(
      <Spinning className='testing' />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
