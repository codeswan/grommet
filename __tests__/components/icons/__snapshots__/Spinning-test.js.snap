exports[`Spinning has correct className rendering 1`] = `
<svg
  className="grommetux-icon-spinning testing"
  role="img"
  version="1.1"
  viewBox="0 0 48 48">
  <title>
    Spinning
  </title>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="20"
    stroke="#ddd"
    strokeDasharray="24px 8px"
    strokeWidth="4" />
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="20"
    stroke="#333"
    strokeDasharray="24px 104px"
    strokeWidth="4" />
</svg>
`;

exports[`Spinning has correct default options 1`] = `
<svg
  className="grommetux-icon-spinning"
  role="img"
  version="1.1"
  viewBox="0 0 48 48">
  <title>
    Spinning
  </title>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="20"
    stroke="#ddd"
    strokeDasharray="24px 8px"
    strokeWidth="4" />
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="20"
    stroke="#333"
    strokeDasharray="24px 104px"
    strokeWidth="4" />
</svg>
`;

exports[`Spinning has correct small=true rendering 1`] = `
<svg
  className="grommetux-icon-spinning grommetux-icon-spinning--small"
  role="img"
  version="1.1"
  viewBox="0 0 48 48">
  <title>
    Spinning
  </title>
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="20"
    stroke="#ddd"
    strokeDasharray="24px 8px"
    strokeWidth="4" />
  <circle
    cx="24"
    cy="24"
    fill="none"
    r="20"
    stroke="#333"
    strokeDasharray="24px 104px"
    strokeWidth="4" />
</svg>
`;
